### PR TITLE
Increase website contrast and simplify layout

### LINKED ISSUE #719: Mobile website homepage hero is squished
This would have changed when we updated our title and slogan. There should be some text and padding tweaks we can make to fix it. 

image

### LINKED ISSUE #830: Make website search more prominent
Right now it's hard to notice the search input, even though it's fairly big. I did some "user testing" and it wasn't obvious to a lot of people. I think we can improve this by increasing the contrast of the placeholder, and adding a magnifying glass icon.

### LINKED ISSUE #831: Allow TOC to be disabled
On some short pages it's distracting and not necessary. Notably on the homepage it makes it feel quite cluttered since we have a lot going on with the blog header.

### PR DESCRIPTION
**Pull request by sebmck**
These are some changes to the website that should increase contrast and make the layout easier to visually scan.

 - Put external link icons at the bottom of the sidebar instead of the top.
 - Add icon and increase contrast of search input, which is now at the very top of the sidebar.
 - Combine philosophy and team page into an "About" page
 - Rename "Core Documentation" category to "Guides"
 - Removed color scheme switcher from the right sidebar.
 - Added new color scheme switcher to the bottom of the left sidebar that uses icons.
 - Made the right sidebar only contain a table of contents (TOC). It is now hidden on pages with less than three subheadings. It no longer includes the top level heading. It is also floating to the side so it does not cause layout movement when moving between TOCless and TOC pages. It stops floating when the viewport shrinks to a certain size.
 - Added URLs to blog post author.
 - Added top margin for netlify badge.
 - Update `pattern-black` to be the same shape as `pattern-white` to reduce color scheme transition jitter.
 - Switch to rgba colors for some color scheme constants. Reduces color scheme transition jitter and keeps contrast consistent.

Fixes #831, fixes #830, fixes #719.

Screen Shot 2020-07-19 at 12 39 05 AM

Screen Shot 2020-07-19 at 12 39 23 AM

### CODE REVIEW

**Comment by sebmck** [comment]
Made some more changes:

 - Removed the sidebar scroll position border JavaScript, and made us do it in CSS instead for a nice smooth scroll without the left sidebar moving around.
 - Hide dark mode selector when JavaScript is disabled.
 - Automatically expand homepage code example when JavaScript is disabled.
 - Use `@media (prefers-color-scheme: dark)` to force dark theme when JavaScript is disabled.

**Comment by sebmck** [comment]
Mobile fixes:

 - Put homepage latest blog post info on the same line.
 - Added `margin-bottom` to netlify badge.
 - On TOCless pages, make the left sidebar in the hamburger menu full width.
 - Use an inset box shadow on mobile, instead of a outline, on the search input focus.

**Comment by EduardoLopes** [comment]
This looks good! 

The color scheme switcher wasn't looking good (too cartoonish; and now i think the same of the "Get started" button), i was about to change it and remove it from the right sidebar, to remove the entire sidebar when not showing the toc. Looks a lot better now!

The main content container is too narrow when there's a toc. I believe this is a Windows only issue, because Windows scaling is set to 1.25 and mess up everything.
Annotation 2020-07-19 130625

There's some issues on mobile, but i'll take care of that today!

File: website/src/blog/posts/test.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ---
 title: This is my first post
 description: This is a post on My Blog about agile frameworks.
-author: sebmck
+author_name: Sebastian Mckenzie
```
Comment (yassere) [inline website/src/blog/posts/test.md:4]: How do you decide when to capitalize the K in your name? 😁

Comment (sebmck) [reply website/src/blog/posts/test.md:4]: Oops, it should always be capitalized! Can't believe I let this happen.

**Pull request merged**